schema/encoding/jsonschema/jsonschema.go

```diff
@@ -18,9 +18,35 @@ var (
 	ErrNotImplemented = errors.New("not implemented")
 )
 
+// Wrap IO writer so we can consolidate error handling
+// in a single place. Also track properties written
+// so we know when to emit a separator.
 type errWriter struct {
-	w   io.Writer
-	err error
+	w          io.Writer // writer instance
+	err        error     // track errors
+	properties int       // track properties written
+}
+
+// comma optionally outputs a comma.
+// Invoke this when you're about to write a property.
+// Tracks how many have been written and emits if not the first.
+func (ew *errWriter) comma() {
+	if ew.properties > 0 {
+		ew.writeString(",")
+	}
+	ew.properties++
+}
+
+func (ew *errWriter) resetPropertiesCount() {
+	ew.properties = 0
+}
```

Comment:
I'm not sure about those two methods on a type called `errWriter`. Also, I don't see the relation between `comma` and `resetPropertiesCount` semantically, those two method names do not show that they are working together.

As you only use this logic in `serializeField`, why don't you just move it inside this method? Actually you are even forced to do that, as I think the current implementation introduce a bug when the following sequence of call happens: `schemaToJSONSchema` > `serializeField` > `validatorToJSONSchema` > `schemaToJSONSchema` > `serializeField`. As you have two `serializeField` in flight at the same time, their states (which is global to the writer) will mess up with each other.